robot/impl/robot_reconfigure_test.go

```diff
@@ -3819,7 +3819,7 @@ func TestResourceConstructCtxCancel(t *testing.T) {
 			conf resource.Config,
 			logger logging.Logger,
 		) (resource.Resource, error) {
-			contructCount++
+			constructCount++
 			wg.Add(1)
 			defer wg.Done()
 			cFunc.c()
```

Comment:
technically not part of this change, but why do we need to introduce a struct that wraps a cancellation function?

Reply:
no, went ahead and refactored the whole test